src/Sign.Core/SignatureProviders/ClickOnceSignatureProvider.cs

```diff
@@ -129,19 +129,26 @@ public async Task SignAsync(IEnumerable<FileInfo> files, SignOptions options)
                             throw new Exception(message);
                         }
 
-                        // Read the publisher name from the manifest for use below
-                        XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
-                        XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
-                        XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
-                        string publisherName = publisherIdentity!.Attribute("name")!.Value;
+                        string publisherParam = "";
 
-                        var publisherParam = "";
-
-                        Dictionary<string, List<string>> dict = DistinguishedNameParser.Parse(publisherName);
-                        if (dict.TryGetValue("CN", out var cns))
+                        if (string.IsNullOrEmpty(options.PublisherName))
+                        {
+                            // Read the publisher name from the manifest for use below
+                            XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
+                            XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
+                            XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
+                            string publisherName = publisherIdentity!.Attribute("name")!.Value;
```

Comment:
Should we replace `"name"` string with `const` on top?

Reply:
A good rule of thumb for this is if the string literal is only used in one place, it's optional.  I didn't use a constant here because `"name"` is only used once, and introducing a constant is of debatable value.  However, as soon as a string literal is used in multiple places, a constant should be used.